packages/deployments/contracts/contracts/core/connext/facets/BridgeFacet.sol

```diff
@@ -207,74 +207,175 @@ contract BridgeFacet is BaseConnextFacet {
-   * @param _sender The sender address
-   * @param _message The message
+   * @notice The interface-compliant entrypoint for nomad relayers. Handles an incoming nomad router message that has
+   * been verified optimistically. Wraps `_reconcile`, which contains the business logic involved in completing the
+   * xchain update.
+   *
+   * @dev Since this method will be called by nomad relayers, it should not consume arbitrary amounts of gas under
+   * any circumstances.
+   *
+   * @param _origin - The origin domain's numeric ID.
+   * @param _nonce - The unique numeric identifier for the message from origin to destination.
+   * @param _sender - The sender identifier.
+   * @param _message - The message bytes.
    */
   function handle(
     uint32 _origin,
     uint32 _nonce,
     bytes32 _sender,
     bytes memory _message
   ) external onlyReplica onlyRemoteRouter(_origin, _sender) {
-    // handle the action
     _reconcile(_origin, _message);
   }
 
   /**
-   * @notice Called on the destination domain to disburse correct assets to end recipient
-   * and execute any included calldata
-   * @dev Can be called prior to or after `handle`, depending if fast liquidity is being
-   * used.
+   * @notice Called on a destination domain to disburse correct assets to end recipient and execute any included
+   * calldata.
+   *
+   * @dev Can be called before or after `handle` [reconcile] is called (regarding the same transfer), depending on
+   * whether the fast liquidity route (i.e. funds provided by routers) is being used for this transfer. As a result,
+   * executed calldata (including properties like `originSender`) may or may not be verified depending on whether the
+   * reconcile has been completed (i.e. the optimistic confirmation period has elapsed).
+   *
+   * @param _args - ExecuteArgs arguments.
+   * @return bytes32 - The transfer ID of the crosschain transfer. Should match the xcall's transfer ID in order for
+   * reconciliation to occur.
    */
-  function execute(ExecuteArgs calldata _args) external whenBridgeNotPaused returns (bytes32) {
+  function execute(ExecuteArgs calldata _args) external whenBridgeNotPaused nonReentrant returns (bytes32) {
     (bytes32 transferId, bool reconciled) = _executeSanityChecks(_args);
 
     // execute router liquidity when this is a fast transfer
-    (uint256 amount, address adopted) = _handleExecuteLiquidity(transferId, !reconciled, _args);
+    // asset will be adopted unless specified to be local in params
+    (uint256 amount, address asset) = _handleExecuteLiquidity(transferId, !reconciled, _args);
 
     // execute the transaction
-    _handleExecuteTransaction(_args, amount, adopted, transferId, reconciled);
+    uint256 amountWithSponsors = _handleExecuteTransaction(_args, amount, asset, transferId, reconciled);
 
     // Set the relayer for this transaction to allow for future claim
     s.transferRelayer[transferId] = msg.sender;
```

Comment:
In line with the other changes where we move state changes up closest to the top of fn as possible, let's move this change to just under `_executeSanityChecks` ?

Reply:
done!